airbyte-cdk/python/airbyte_cdk/sources/declarative/retrievers/simple_retriever.py

```diff
@@ -372,11 +373,13 @@ def read_records(
             stream_slice,
             stream_state,
         )
-        for record in records_generator:
-            # Only record messages should be parsed to update the cursor which is indicated by the Mapping type
-            if isinstance(record, Mapping):
-                self.stream_slicer.update_cursor(stream_slice, last_record=record)
-            yield record
+        first, records_generator = self._peek(records_generator)
```

Comment:
BTW, why not just:

```python
cursor_updated = False
for record in records_generator:
    self.stream_slicer.update_cursor(stream_slice, last_record=record)
    cursor_updated = True
    yield record
if not cursor_updated:
    last_record = self._last_records[-1] if self._last_records else None
    self.stream_slicer.update_cursor(stream_slice, last_record=last_record)
```

Reply:
this works too. updated to use this approach since it removes the need for the additional function.